nix/open-webui.nix

```diff
@@ -0,0 +1,95 @@
+# Based on: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/open-webui.nix
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    enable = lib.mkEnableOption "Open-WebUI server";
+    package = lib.mkPackageOption pkgs "open-webui" { };
+
+    dataDir = lib.mkOption {
+      type = types.str;
+      default = "./data/${name}";
+      description = "The Open-WebUI data directory";
+    };
+
+    host = lib.mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      example = "0.0.0.0";
+      description = ''
+        The host address which the Open-WebUI server HTTP interface listens to.
+      '';
+    };
+
+    port = lib.mkOption {
+      type = types.port;
+      default = 1111;
+      example = 11111;
+      description = ''
+        Which port the Open-WebUI server listens to.
+      '';
+    };
+
+    environment = lib.mkOption {
+      type = types.attrsOf types.str;
+      default = {
+        SCARF_NO_ANALYTICS = "True";
+        DO_NOT_TRACK = "True";
+        ANONYMIZED_TELEMETRY = "False";
+      };
+      example = ''
+        {
+          OLLAMA_API_BASE_URL = "http://127.0.0.1:11434";
+          # Disable authentication
+          WEBUI_AUTH = "False";
+        }
+      '';
+      description = "Extra environment variables for Open-WebUI";
+    };
+
+    outputs.settings = lib.mkOption {
+      type = types.deferredModule;
+      internal = true;
+      readOnly = true;
+      default = {
+        processes = {
+          "${name}" = {
+            inherit (config) environment;
+            command = pkgs.writeShellApplication {
+              name = "open-webui-wrapper";
+              runtimeInputs = [ pkgs.open-webui ];
+              text = ''
+                set -x
+                if [ ! -d ${config.dataDir} ]; then
+                  mkdir -p ${config.dataDir} 
+                fi
+
+                DATA_DIR=$(readlink -f ${config.dataDir})
+                STATIC_DIR=$DATA_DIR
+                HF_HOME=$DATA_DIR
+                SENTENCE_TRANSFORMERS_HOME=$DATA_DIR
```

Comment:
Is there some kind of attribute (a bit like with systemd) where we can define the environment variables?

Reply:
Apparently it is possible, see: https://github.com/juspay/services-flake/blob/c9e553879a792b9a746f0b8395c9dbffd58d8342/example/share-services/pgweb/flake.nix#L32C1-L32C118

How about moving the environment variable out of the shell script and move them in there?

Reply:
excellent now!

Reply:
Thanks @drupol! Did it over here: https://github.com/juspay/services-flake/pull/219/commits/72480e26c051930b449b6bb6c3b88e3b4161fa48

Now it is similar to what `systemd` allows.